File: vapor-app/src/main/java/com/tevinjeffrey/vapor/ui/files/fragments/presenters/FilesPresenter.java
package com.tevinjeffrey.vapor.ui.files.fragments.presenters;

import com.tevinjeffrey.vapor.ui.base.Presenter;
import com.tevinjeffrey.vapor.ui.base.View;

public interface FilesPresenter<V extends View> extends Presenter<V> {
    void loadData(boolean pullToRefresh, boolean refreshData, boolean useCursor);
    boolean isLoading();
}

File: vapor-app/src/main/java/com/tevinjeffrey/vapor/ui/files/fragments/presenters/FilesPresenter.kt
package com.tevinjeffrey.vapor.ui.files.fragments.presenters

import com.tevinjeffrey.vapor.ui.base.Presenter
import com.tevinjeffrey.vapor.ui.base.View

interface FilesPresenter<V : View> : Presenter<V> {
    fun loadData(pullToRefresh: Boolean, refreshData: Boolean, useCursor: Boolean)
    val isNotLoading: Boolean
    fun shouldShowEmpty(): Boolean
}
